Tasks/ServiceFabricUpdateAppVersions/Update-PackageVersion.psm1

```diff
@@ -77,7 +77,7 @@
                     # Search the package files for differences (Unless the user doesn't want to log all changes and we already know the package xml changed)
                     $newPackagePath = Join-Path $NewPackageRoot $packageName
                     $oldPackagePath = Join-Path $OldPackageRoot $packageName
-                    if (Find-FileChanges $newPackagePath $oldPackagePath $LogIndent -LogAllChanges:$LogAllChanges)
+                    if ((Test-Path $newPackagePath) -and (Find-FileChanges $newPackagePath $oldPackagePath $LogIndent -LogAllChanges:$LogAllChanges))
```

Comment:
Excellent point, @NCarlsonMSFT.  So it would seem we should update the version for Scenario 3 just to be safe.  It will account for your scenario and shouldn't be a problem for the "normal" scenario.